napari/_qt/layer_controls/_tests/test_qt_layer_controls.py

```diff
@@ -79,3 +80,78 @@ def test_set_text_then_set_visible_updates_checkbox(
     layer.text.visible = True
 
     assert ctrl.textDispCheckBox.isChecked()
+
+
+# The following tests handle changes to the layer's visible and
+# editable state for layer control types that have controls to edit
+# the layer. For more context see:
+# https://github.com/napari/napari/issues/1346
+EDITABLE_TEST_LAYERS = (
+    Labels(np.zeros((3, 4), dtype=int)),
+    Points(np.empty((0, 2))),
+    Shapes(np.empty((0, 2, 4))),
+)
```

Comment:
Are we sure that using the global instances of layers and sharing them between tests is a good idea? It may lead to a situation when running individual tests will lead to failing, and running the whole file will pass. 
And it will make debugging harder.

Reply:
Yeah, I didn't like this either. The simplest fix is to store the type and data, then initialize in the test so that we get a new instance each time. There does seem to be a way to do this with fixtures, so that it would something the following:

```python
@pytest.fixture(name="editable_layer", params=(
    (Labels, np.zeros((3, 4), dtype=int)),
    (Points, np.empty((0, 2))),
    (Shapes, np.empty((0, 2, 4))),
))
def _editable_layer(request):
    LayerType, data = request.param
    return LayerType(data)

def test_make_visible_when_editable_enables_edit_buttons(qtbot, editable_layer):
    editable_layer.editable = True
    editable_layer.visible = False
    controls = make_layer_controls(qtbot, editable_layer)
    assert_no_edit_buttons_enabled(controls)

    editable_layer.visible = True

    assert_all_edit_buttons_enabled(controls)
```

What do you think?

Reply:
FYI, the source for the above approach is https://hynek.me/til/parametrized-pytest-fixtures/ (from a PSF fellow). I can't find the pytest docs on this specific usage.

Reply:
Looks good. I du not understand of using `_` and set the fixture name. This part looks obsolete.

Reply:
Thanks for the link - somehow I read that (or some version of it) multiple times without understanding it, but now it seems obvious. The use of `_` and `name` is gone.

I parametrized the fixture instead of the test because things are fairly simple here and it keeps the tests concise.